libyara/atoms.c

```diff
@@ -161,8 +161,8 @@ int yr_atoms_heuristic_quality(YR_ATOMS_CONFIG* config, YR_ATOM* atom)
         // than the rest. We want to favor atoms that contain bytes outside
         // those ranges because they generate less additional atoms during
         // calls to _yr_atoms_case_combinations.
-        if (yr_lowercase[atom->bytes[i]] >= 'a' &&
-            yr_lowercase[atom->bytes[i]] <= 'z')
+        if ((yr_lowercase[atom->bytes[i]] >= 'a' && yr_lowercase[atom->bytes[i]] <= 'z') ||
+        (yr_lowercase[atom->bytes[i]] >= 'A' && yr_lowercase[atom->bytes[i]] <= 'Z'))
```

Comment:
This change has no effect because `yr_lowercase[atom->bytes[i]]` will never be in the range `A..Z`. The lookup table `yr_lowercase` converts any `atom->bytes[i]` in the range `A..Z` to its lowercase counterpart. In 25f576adf3094061fe5c6ba87c103b460ecb6171 I've added your test case to the original code, and it passes without any other changes.

Reply:
Thanks @plusvic, sorry I hadn't noticed that `yr_lowercase` routine!